Ginger/Ginger/ALM/Repository/QtestRepository.cs

```diff
@@ -0,0 +1,405 @@
+
+                foreach (QtestSuiteTreeItem testSetItemtoImport in testSuitesItemsToImport)
+                {
+                    try
+                    {
+                        Reporter.ToStatus(eStatusMsgKey.ALMTestSetImport, null, testSetItemtoImport.Name);
+                        QtestTestSuite TS = new QtestTestSuite();
+                        TS.ID = testSetItemtoImport.ID;
+                        TS.Name = testSetItemtoImport.Name;
+                        TS = ((QtestCore)ALMIntegration.Instance.AlmCore).ImportTestSetData(TS);
+
+                        //convert test set into BF
+                        BusinessFlow tsBusFlow = ((QtestCore)ALMIntegration.Instance.AlmCore).ConvertQCTestSetToBF(TS);
+
+                        if ( WorkSpace.Instance.Solution.MainApplication != null)
+                        {
+                            //add the applications mapped to the Activities
+                            foreach (Activity activ in tsBusFlow.Activities)
+                                if (string.IsNullOrEmpty(activ.TargetApplication) == false)
+                                    if (tsBusFlow.TargetApplications.Where(x => x.Name == activ.TargetApplication).FirstOrDefault() == null)
+                                    {
+                                        ApplicationPlatform appAgent =  WorkSpace.Instance.Solution.ApplicationPlatforms.Where(x => x.AppName == activ.TargetApplication).FirstOrDefault();
+                                        if (appAgent != null)
+                                            tsBusFlow.TargetApplications.Add(new TargetApplication() { AppName = appAgent.AppName });
+                                    }
+                            //handle non mapped Activities
+                            if (tsBusFlow.TargetApplications.Count == 0)
+                                tsBusFlow.TargetApplications.Add(new TargetApplication() { AppName =  WorkSpace.Instance.Solution.MainApplication });
+                            foreach (Activity activ in tsBusFlow.Activities)
+                            {
+                                if (string.IsNullOrEmpty(activ.TargetApplication))
+                                    activ.TargetApplication = tsBusFlow.MainApplication;
+                                activ.Active = true;
+                            }
+                        }
+                        else
+                        {
+                            foreach (Activity activ in tsBusFlow.Activities)
+                                activ.TargetApplication = null; // no app configured on solution level
+                        }
+
+                        //save bf
+                        WorkSpace.Instance.SolutionRepository.AddRepositoryItem(tsBusFlow);                        
+                        Reporter.HideStatusMessage();
+                    }
+                    catch (Exception ex)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)